[docs] Updated direnv docs to reflect update after v0.5.0 (#1080)
jetpack-io/devbox: docs/app/docs/ide_configuration/direnv.md

#### New Project

If you have direnv installed, Devbox will generate an .envrc file when you run `devbox generate direnv` and enables it by running `direnv allow` in the background:

```bash
➜  devbox generate direnv
Success: generated .envrc file
Success: ran `direnv allow`
direnv: loading ~/src/devbox/docs/.envrc
direnv: using devbox
```

> This will generate a `.envrc` file in your project directory along with your `devbox.json`, if you make changes to `.envrc` file, you can run `direnv allow` so that your shell will activate whenever you navigate to the directory or `direnv revoke` to stop. Any changes to `devbox.json` will automatically trigger direnv to re-setup the environment.

Reviewer: ```suggestion
This will generate a `.envrc` file in your project directory that contains `devbox.json`. Run `direnv allow` to activate your shell upon directory navigation. Run `direnv revoke` to stop. Changes to `devbox.json` automatically trigger direnv to reset the environment.
```

prefix = [ "/absolute/path/to/project" ]

```

### VSCode setup with direnv

> Best experience to have VSCode integration with a direnv environment is to open VSCode from a terminal window where direnv is activated (`code .`). Alternatively, there is [direnv VSCode extension](https://marketplace.visualstudio.com/items?itemName=mkhl.direnv) that helps integrate VSCode's environment if the VSCode workspace has a `.envrc` file.

Reviewer: ```suggestion
To seamlessly integrate VSCode with a direnv environment, follow these steps:

1. Open a terminal window and activate direnv with `direnv allow`.
2. Launch VSCode from the same terminal window using the command `code .` This ensures that VSCode inherits the direnv environment.

Alternatively, you can use the [direnv VSCode extension](https://marketplace.visualstudio.com/items?itemName=mkhl.direnv) if your VSCode workspace has a .envrc file.
```
